applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+from KratosMultiphysics import *
+from KratosMultiphysics.FluidDynamicsApplication import *
+try:
+    from KratosMultiphysics.ExternalSolversApplication import *
+except ImportError:
+    pass
+
+class FluidDynamicsAnalysis(object):
+    '''Main script for fluid dynamics simulations using the navier_stokes family of python solvers.'''
+
+    def __init__(self,parameter_file_name='ProjectParameters.json'):
+        super(FluidDynamicsAnalysis,self).__init__()
+
+        with open(parameter_file_name,'r') as parameter_file:
+            self.project_parameters = Parameters( parameter_file.read() )
+
+        self.echo_level = self.project_parameters["problem_data"]["echo_level"].GetInt()
+        self.parallel_type = self.project_parameters["problem_data"]["parallel_type"].GetString()
+
+        # If this is an MPI run, load the distributed memory modules
+        if (self.parallel_type == "MPI"):
+            from KratosMultiphysics.mpi import mpi
+            import KratosMultiphysics.MetisApplication as Metis
+            import KratosMultiphysics.TrilinosApplication as Trilinos
+            self.is_printing_rank = (mpi.rank == 0)
+        else:
+            self.is_printing_rank = True
+        
+    def SetUpModel(self):
+        '''Initialize the model part for the problem (stored as self.model_part) and other general model data.'''
+
+        model_part_name = self.project_parameters["problem_data"]["model_part_name"].GetString()
+        self.main_model_part = ModelPart(model_part_name)
+
+        self.domain_size = self.project_parameters["problem_data"]["domain_size"].GetInt()
+        self.main_model_part.ProcessInfo.SetValue(DOMAIN_SIZE, self.domain_size)
+
+        ## Solver construction
+        import python_solvers_wrapper_fluid
+        self.solver = python_solvers_wrapper_fluid.CreateSolver(self.main_model_part, self.project_parameters)
+
+        self._SetUpRestart()
+
+        if self.load_restart:
+            self.restart_utility.LoadRestart()
+        else:
+            self.solver.AddVariables()
```

Comment:
I think variables and dofs can be added in either case

It is internally checked if they exist already